app/src/main/java/com/example/parkingsystem/mvp/view/ParkingSpaceReservationView.java

```diff
@@ -78,26 +79,44 @@ public void enableButtonEnd() {
 
     @Override
     public int getParkingSpace() {
-        return Integer.parseInt(binding.inputParkingSpaceReservationPlace.getText().toString());
+        String parkingSpace = binding.inputParkingSpaceReservationPlace.getText().toString();
+        if (parkingSpace.isEmpty()) {
+            return Constants.NUMBER_MINUS_ONE;
+        }
+        return Integer.parseInt(parkingSpace);
     }
 
     @Override
     public int getSecurityCode() {
-        return Integer.parseInt(binding.inputParkingSpaceReservationCode.getText().toString());
+        String securityCode = binding.inputParkingSpaceReservationCode.getText().toString();
+        if (securityCode.isEmpty()) {
+            return Constants.NUMBER_MINUS_ONE;
+        }
+        return Integer.parseInt(securityCode);
     }
 
     @Override
     public void showSaveDone() {
         Context context = getContext();
         Activity activity = getActivity();
-        String message = null;
+        String message;
         if (context != null && activity != null) {
             message = context.getString(R.string.toast_parking_space_reservation_save);
             showMessage(context, message);
             activity.finish();
         }
     }
 
+    @Override
+    public void showReleasedPastReservations(int amountReservations) {
+        Context context = getContext();
+        String message;
```

Comment:
this message variable is not necessary i think

Reply:
Changed!